### PR TITLE
add egale eye networks fingerprint

### PR DESCRIPTION
**Pull request by ssikdar-r7**
## Description
This PR adds HTML TITLE fingerprints for Eagle Eye Networks  for their Cloud Video Surveillance System. ( https://www.een.com/product/cloud-vms-system-overview/ )

## Motivation and Context

For example for `/`:

```
$ curl -sk  https://<redacted host>/ 
<!doctype html>

<html lang="en">
<head>
    <meta charset="utf-8">

    <title>Web Local Display Unauthorized</title>
    <meta name="description" content="Bridge local login error">
    <meta name="author" content="EagleEyeNetworks">
    <link rel="icon" href="_images/favicon.ico" />
    <link rel="stylesheet" href="_css/error_style.css">
    <link rel="stylesheet" href="_css/bootstrap.min.css">
</head>

<body>
    <div align="center" class="container">
         <img src="_images/logo_large.png"/>
         <hr>
         <p class="red">The user name and password that you have entered are incorrect.</p>
    </div> 
</body>
</html>
```

Similarly for `index.html`:
```
$ curl -sk  https://<redacted host>/index.html | perl -l -0777 -ne 'print $1 if /<title.*?>\s*(.*?)\s*<\/title/si'
Eagle Eye Networks (Local)
```

## How Has This Been Tested?
```
$ curl -sk  https://<redacted host>/index.html | perl -l -0777 -ne 'print $1 if /<title.*?>\s*(.*?)\s*<\/title/si' | ./bin/recog_match ./xml/html_title.xml
MATCH: {"matched"=>"EagleEyeNetworks to manage IoT cameras home webpage", "hw.product"=>"Cloud Video Surveillance System", "hw.vendor"=>"Eagle Eye Networks", "service.protocol"=>"", "fingerprint_db"=>"html_title", "data"=>"Eagle Eye Networks (Local)"}
```
```
$ curl -sk  https://<redacted host>/ | perl -l -0777 -ne 'print $1 if /<title.*?>\s*(.*?)\s*<\/title/si' | ./bin/recog_match ./xml/html_title.xml
MATCH: {"matched"=>"EagleEyeNetworks to manage IoT cameras unauthorized webpage", "hw.product"=>"Cloud Video Surveillance System", "hw.vendor"=>"Eagle Eye Networks", "service.protocol"=>"", "fingerprint_db"=>"html_title", "data"=>"Web Local Display Unauthorized"}
```
## Types of changes

- New feature (non-breaking change which adds functionality)
( its a change to `html_title.xml` )

## Checklist:

- [ not required] I have updated the documentation accordingly (or changes are not required).
- [not required ] I have added tests to cover my changes (or new tests are not required).
- [X ] All new and existing tests passed.

```
9 scenarios (9 passed)
20 steps (20 passed)
0m12.480s
```

### CODE REVIEW

File: xml/html_title.xml [hunk 1]
```diff
@@ -331,4 +331,16 @@
     <param pos="0" name="os.product" value="Linux"/>
     <param pos="0" name="os.cpe23" value="cpe:/o:debian:debian_linux:-"/>
   </fingerprint>
+  <fingerprint pattern="^Eagle Eye Network">
+    <description>EagleEyeNetworks to manage IoT cameras unauthorized webpage</description>
+    <example>Eagle Eye Networks (Local)</example>
+    <param pos="0" name="service.product" value="Cloud Video Surveillance System"/>
```
Comment (jhart-r7) [inline xml/html_title.xml]: Why service?  I think `hw` is more appropriate here -- the service is likely nginx/apache/whatever and not something eagle-eye specific.

File: xml/html_title.xml [hunk 2]
```diff
@@ -331,4 +331,16 @@
...
+    <param pos="0" name="service.vendor" value="Eagle Eye Networks"/>
```
Comment (jhart-r7) [inline xml/html_title.xml]: also, you could possibly add a `<param_type>.device` param here too, maybe something camera related.

File: xml/html_title.xml (hunk 1 above)
Comment (ssikdar-r7) [reply xml/html_title.xml]: addressed in https://github.com/rapid7/recog/pull/235/commits/41ea23dc54f6c0553c03c8055d97bfce7d0a0da7

File: xml/html_title.xml (hunk 2 above)
Comment (ssikdar-r7) [reply xml/html_title.xml]: addressed in https://github.com/rapid7/recog/pull/235/commits/0bd2353efda09e54e8afe564513701b481c9344c

File: xml/html_title.xml [hunk 3]
```diff
@@ -331,4 +331,18 @@
...
+    <param pos="0" name="hw.product" value="Cloud Video Surveillance System"/>
+    <param pos="0" name="hw.vendor" value="Eagle Eye Networks"/>
+    <param pos="0" name="hw.device" value="Eagle Eye Bridge"/>
+  </fingerprint>
+  <fingerprint pattern="^Web Local Display Unauthorized$">
+    <description>EagleEyeNetworks to manage IoT cameras unauthorized webpage</description>
+    <example>Web Local Display Unauthorized</example>
+    <param pos="0" name="hw.product" value="Cloud Video Surveillance System"/>
+    <param pos="0" name="hw.vendor" value="Eagle Eye Networks"/>
+    <param pos="0" name="hw.device" value="Eagle Eye Bridge"/>
```
Comment (jhart-r7) [inline xml/html_title.xml]: `.device` is meant to describe the general type of device that this is in a non-vendor specific way.  so, printer, camera, laptop, etc, are all good `.device` types.  Perhaps this should be the `.product`?

Comment (ssikdar-r7) [reply xml/html_title.xml]: https://github.com/rapid7/recog/pull/235/commits/99a309688194ae1e21b46efe37ee1c0cfbbe2659

File: xml/html_title.xml [hunk 4]
```diff
@@ -331,4 +331,16 @@
     <param pos="0" name="os.product" value="Linux"/>
     <param pos="0" name="os.cpe23" value="cpe:/o:debian:debian_linux:-"/>
   </fingerprint>
+  <fingerprint pattern="^Eagle Eye Network">
+    <description>EagleEyeNetworks to manage IoT cameras home webpage</description>
+    <example>Eagle Eye Networks (Local)</example>
+    <param pos="0" name="hw.product" value="Eagle Eye Bridge"/>
```
Comment (jhart-r7) [inline xml/html_title.xml]: I believe the product is called just Bridge.  See https://www.een.com/product/bridge-models/, where in the tables near the bottom they call the devices Bridge or CMVR.

**Pull request merged**